src/Lecture1.hs

```diff
@@ -119,4 +145,18 @@ and lower than 6 elements (4, 5, 6, 7, 8 and 9).
 
 🕯 HINT: Use recursion to implement this function.
 -}
-lowerAndGreater n list = error "TODO"
+getNums :: Int -> [Int] -> [Int] -> [Int]
+getNums n result list
+    | null list = result
+    | ((head list) > n) = getNums n ((head result + 1):[] ++ (tail result)) (tail list)
+    | (head list) == n = getNums n result (tail list)
+    | (head list) < n = getNums n (init result ++ (last result + 1):[]) (tail list)
```

Comment:
GHC produces warnings about non-exhaustive patterns. It happens here because GHC cannot check that your conditions actually cover all possible cases. For this, you need to have the last guard under the `otherwise` check 💂🏻